src/ray/protobuf/instance_manager.proto

```diff
@@ -0,0 +1,82 @@
+
+// Represents the state of a launched instance.
+// An instance is considered launched as long as
+// it has a unique instance_id associated with it.
+//
+// Note a launched instance may be DEAD. In this case,
+// the state will be garbage collected after some timeout
+// period (by default 30 minutes).
+message LaunchedInstance {
+  enum LaunchedInstanceState {
+    ALIVE = 1;
+    DEAD = 2;
+    PREMEPTING = 3;
+    TERMINATING = 4;
+  }
+  bytes instance_id = 1;
+  string node_type = 2;
+  LaunchedInstanceState instance_state = 3; 
+}
+
+// Representing a pending request to launch an instance.
+//
+// Note a request could be failed due to various reasons.
+// The detailed reqson could be queried through description
+// field. A failed launching request will be GCed
+// after some timeout period (by default 30 minutes).
+message LauchingInstance {
+  enum LauchingRequestState {
+    QUEUED = 1;
+    REQUESTED = 2;
```

Comment:
What's the difference between QUEUED and REQUESTED, and what does it mean to have RESOURCES_ALLOCATED?

The API contract has to be extremely precise here.